Content.Server/Internals/InternalsProviderUI.cs

```diff
@@ -0,0 +1,108 @@
+        {
+            base.Initialize();
+
+            SubscribeLocalEvent<InternalsProviderUIComponent, ComponentStartup>(OnStartup);
+            SubscribeLocalEvent<InternalsProviderUIComponent, ActivateInWorldEvent>(OnActivate);
+            SubscribeLocalEvent<InternalsProviderUIComponent, UseInHandEvent>(OnUse);
+            SubscribeLocalEvent<InternalsProviderUIComponent, InteractHandEvent>(OnInteract);
+            SubscribeLocalEvent<InternalsProviderUIComponent, GasTankPressureDeficitEvent>(OnTankPressureChange);
+        }
+
+        public void OnStartup(EntityUid uid, InternalsProviderUIComponent component, ComponentStartup args)
+        {
+            component.UserInterface = component.Owner.GetUIOrNull(SharedGasTankUiKey.Key);
+            if (component.UserInterface != null)
+            {
+                component.UserInterface.OnReceiveMessage += (msg) => UserInterfaceOnOnReceiveMessage(component, msg);
+            }
+        }
+
+        public void OnActivate(EntityUid uid, InternalsProviderUIComponent component, ActivateInWorldEvent args)
+        {
+            if (!args.User.TryGetComponent(out ActorComponent? actor)) return;
+            OpenInterface(component, actor.PlayerSession);
+        }
+        public void OnUse(EntityUid uid, InternalsProviderUIComponent component, UseInHandEvent args)
+        {
+            if (!args.User.TryGetComponent(out ActorComponent? actor)) return;
+            OpenInterface(component, actor.PlayerSession);
+        }
+
+        public void OnInteract(EntityUid uid, InternalsProviderUIComponent component, InteractHandEvent args)
+        {
+            if (!args.User.TryGetComponent(out ActorComponent? actor)) return;
+            OpenInterface(component, actor.PlayerSession);
+        }
+
+        public void OnTankPressureChange(EntityUid uid, InternalsProviderUIComponent component, GasTankPressureDeficitEvent args)
+        {
+            UpdateUserInterface(component, true);
+        }
+
+        public void OpenInterface(InternalsProviderUIComponent component, IPlayerSession session)
+        {
+            component.UserInterface?.Open(session);
+            UpdateUserInterface(component, true);
+        }
+
+        public void UpdateUserInterface(InternalsProviderUIComponent component, bool initialUpdate = false)
+        {
+            var provider = component.Owner.GetComponent<InternalsProviderComponent>();
+            var internals = provider.Internals;
+
+            var gasTank = component.Owner.GetComponent<GasTankComponent>();
+
+            component.UserInterface?.SetState(
+                new GasTankBoundUserInterfaceState
+                {
+                    TankPressure = gasTank.Air?.Pressure ?? 0,
+                    OutputPressure = initialUpdate ? gasTank.OutputPressure : null,
+                    InternalsConnected = internals != null,
+                    CanConnectInternals = true // todo - fix?
+                }
+            );
+        }
```

Comment:
As above use the EntityManager methods also don't GetComponent unless you're 100% sure the comp exists because it can throw (in this case you're not).